examples/example.hpp

```diff
@@ -581,37 +742,75 @@ class window
     void show(const std::map<int, rs2::frame> frames)
     {
         // Render openGl mosaic of frames
-         if (frames.size())
-         {
-             int cols = int(std::ceil(std::sqrt(frames.size())));
-             int rows = int(std::ceil(frames.size() / static_cast<float>(cols)));
-
-             float view_width = float(_width / cols);
-             float view_height = float(_height / rows);
-             int stream_no =0;
-             for (auto& frame : frames)
-             {
-                 rect viewport_loc{ view_width * (stream_no % cols), view_height * (stream_no / cols), view_width, view_height };
-                 show(frame.second, viewport_loc);
-                 stream_no++;
-             }
-         }
-         else
-         {
-             _main_win.put_text("Connect one or more Intel RealSense devices and rerun the example",
-                 0.4f, 0.5f, { 0.f,0.f, float(_width) , float(_height) });
-         }
-    }
-
-    operator GLFWwindow*() { return win; }
+        if (frames.size())
+        {
+            int cols = int(std::ceil(std::sqrt(frames.size())));
+            int rows = int(std::ceil(frames.size() / static_cast<float>(cols)));
+
+            float view_width = float(_width / cols);
+            float view_height = float(_height / rows);
+            unsigned int stream_no = 0;
+            for (auto& frame : frames)
+            {
+                rect viewport_loc{ view_width * (stream_no % cols), view_height * (stream_no / cols), view_width, view_height };
+                show(frame.second, viewport_loc);
+                stream_no++;
+            }
+        }
+        else
+        {
+            _main_win.put_text("Connect one or more Intel RealSense devices and rerun the example",
+                0.4f, 0.5f, { 0.f,0.f, float(_width) , float(_height) });
+        }
+    }
+
+    //gets as argument a map of the -need to be drawn- frames with their tiles properties,
+    //which indicates where and what size should the frame be drawn on the canvas 
+    void show(const map_of_frames_and_tiles_properties frames)
```

Comment:
1. pass by ref : const map_of_frames_and_tiles_properties& frames
2. The name `map_of_frames_and_tiles_properties` is too long - consider `frames_mosaic` or similar